module/userProjects/helper.js

```diff
@@ -1742,173 +1844,246 @@ module.exports = class UserProjectsHelper {
+				const mongooseIdData = this.mongooseIdData(schemas['projects'].schema)
 
+				//Fetch user profile information by calling user service.
 				let userProfile = await projectService.profileRead(userToken)
-				// Check if the user profile fetch was successful
+				// // Check if the user profile fetch was successful
 				if (!userProfile.success) {
 					throw {
 						message: CONSTANTS.apiResponses.USER_DATA_FETCH_UNSUCCESSFUL,
 						status: HTTP_STATUS_CODE.bad_request.status,
 					}
 				}
-				if (userProfile.success && userProfile.data) {
-					createProject.userProfile = userProfile.data
-				}
 
-				let projectData = await _projectData(data)
-				if (projectData && projectData.success == true) {
-					createProject = _.merge(createProject, projectData.data)
-				} else {
-					return reject(projectData)
+				let createNewProgram = false
+				if (data.program._id && data.program._id !== '') {
+					createNewProgram = false
+				} else if (data.program.name && data.program.name != '') {
+					createNewProgram = true
 				}
 
-				// let createNewProgramAndSolution = false;
-
-				// if (data.programId && data.programId !== "") {
-				//     createNewProgramAndSolution = false;
-				// }
-				// else if (data.programName) {
-				//     createNewProgramAndSolution = true;
-				// }
-
-				if (data.entityId) {
-					// let entityInformation = await entitiesService.entityDocuments({"_id" : data.entityId},"all",userToken)
-					let entityInformation = await _entitiesInformation([data.entityId])
-
-					if (!entityInformation.success) {
-						return resolve(entityInformation)
+				// Fetch program details
+				if (!createNewProgram) {
+					const program = await programQueries.programsDocument(
+						{
+							_id: data.program._id,
+						},
+						'all'
+					)
+					if (!program || !(Object.keys(program).length > 0)) {
+						throw {
+							success: false,
+							message: CONSTANTS.apiResponses.PROGRAM_NOT_FOUND,
+						}
 					}
+				}
 
-					createProject['entityInformation'] = entityInformation.data[0]
-					createProject.entityId = entityInformation.data[0]._id
+				// Throw error if project array is not present in the req.body
+				if (!data.projects || !(data.projects.length > 0)) {
+					throw {
+						success: false,
+						status: HTTP_STATUS_CODE.bad_request.status,
+						message: CONSTANTS.apiResponses.PROJECT_DATA_NOT_FOUND,
+					}
 				}
-				// if (createNewProgramAndSolution) {
 
-				//     let programAndSolutionInformation =
-				//         await this.createProgramAndSolution(
-				//             data.programId,
-				//             data.programName,
-				//             createProject.entityId ? [createProject.entityId] : "",
-				//             userToken
-				//         );
+				let programAndMultipleSolutionsData
+				const projectManatoryFields = ['duration', 'endDate', 'source', 'startDate', 'status', 'tasks', 'title']
+				const tasksMandatoryFields = ['name', 'isDeletable']
```

Comment:
@Prajwal17Tunerlabs even if isDeletable is not passed we will make it false. for griteorks we gave it as mandatory field. if it is not there we can manage. will be useful for elevate project

Reply:
got it, i have removed isDeletable from the mandatory list